src/test/java/gov/healthit/chpl/aqa/asserts/ProductHistoryEyeAsserts.java

```diff
@@ -0,0 +1,45 @@
+package gov.healthit.chpl.aqa.asserts;
+
+import static org.testng.Assert.assertTrue;
+
+import cucumber.api.java.en.Then;
+import gov.healthit.chpl.aqa.pageObjects.ProductHistoryEyePage;
+import gov.healthit.chpl.aqa.stepDefinitions.Base;
+
+/**
+ * Class ProductHistoryEyeAsserts definition.
+ */
+public class ProductHistoryEyeAsserts extends Base {
+
+    /**
+     * Assert that expected developer edit details is displayed on product history eye.
+     * @param expectedDeveloperEditDetails is "Developer changed from"
+     */
+    @Then("^I see the product history of developer edit is recorded as \"([^\"]*)\"$")
+    public void iSeeProductHistoryOfDeveloper(final String expectedDeveloperEditDetails) {
+        String actualDeveloperEditDetails = ProductHistoryEyePage.productHistoryDeveloper(getDriver()).getText();
+        assertTrue((actualDeveloperEditDetails.startsWith(expectedDeveloperEditDetails)));
+    }
+
+    /**
+     * Assert that expected version edit details is displayed on product history eye.
+     * @param expectedVersionEditDetails is "Version changed from"
+     */
+    @Then("^I see the product history of version edit is recorded as \"([^\"]*)\"$")
+    public void iSeeProductHistoryOfVersion(final String expectedVersionEditDetails) {
+        String actualVersionEditDetails = ProductHistoryEyePage.productHistoryVersion(getDriver()).getText();
+        assertTrue((actualVersionEditDetails.startsWith(expectedVersionEditDetails)));
+    }
+
+    /**
+     * Assert that expected product edit details is displayed on product history eye.
+     * @param expectedProductEditDetails is "Product changed from"
+     */
+    @Then("^I see the product history of product edit is recorded as \"([^\"]*)\"$")
+    public void iSeeProductHistoryOfProduct(final String expectedProductEditDetails) {
+        String actualProductEditDetails = ProductHistoryEyePage.productHistoryProduct(getDriver()).getText();
+        assertTrue((actualProductEditDetails.startsWith(expectedProductEditDetails)));
```

Comment:
Shouldn't use the text you're looking for as the data to use to find a PageObject. If the Page changes, the PageObject should be the only thing that needs to change, but here you'd need to change the assert as well. Also, if this test works the first time, but fails to do the edit the next day, it'll still find that history in the eye, when it should report a failure

Reply:
I think only asserting partial text isn't enough as there might be existing history on product that includes that text or like Andrew said above, from previous day's edit. A good check would be to combine the text assert with date validation in same row in 'date' column  when activity happened